bitcoin/src/sighash.rs

```diff
@@ -1378,74 +1376,148 @@ mod tests {
         assert_eq!(expected, hash.into_inner());
     }
 
+    #[cfg(feature = "serde")]
     #[test]
     fn bip_341_sighash_tests() {
-        let data = bip_341_read_json();
-        assert!(data["version"].as_u64().unwrap() == 1u64);
-        let secp = &secp256k1::Secp256k1::new();
-        let key_path = &data["keyPathSpending"].as_array().unwrap()[0];
-
-        let raw_unsigned_tx =
-            hex_decode!(Transaction, key_path["given"]["rawUnsignedTx"].as_str().unwrap());
-        let mut utxos = vec![];
-        for utxo in key_path["given"]["utxosSpent"].as_array().unwrap() {
-            let spk = hex_script!(utxo["scriptPubKey"].as_str().unwrap());
-            let amt = utxo["amountSats"].as_u64().unwrap();
-            utxos.push(TxOut { value: amt, script_pubkey: spk });
+        fn sighash_deser_numeric<'de, D>(deserializer: D) -> Result<SchnorrSighashType, D::Error> where D: actual_serde::Deserializer<'de> {
+            use actual_serde::de::{Deserialize, Error, Unexpected};
+
+            let raw = u8::deserialize(deserializer)?;
+            SchnorrSighashType::from_consensus_u8(raw)
+                .map_err(|_| D::Error::invalid_value(Unexpected::Unsigned(raw.into()), &"number in range 0-3 or 0x81-0x83"))
+        }
+            
+        use crate::hashes::hex::ToHex;
+        use crate::taproot::{TapTweakHash, TapBranchHash};
+        use secp256k1::{self, SecretKey, XOnlyPublicKey};
+        use crate::consensus::serde as con_serde;
```

Comment:
This PR introduces formatting issues. I'm totally fine to personally remember this and fix it later.